packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -86,19 +95,48 @@ function FlowsheetDataGrid(props) {
     rowHeight,
     onCellSelect,
     onClearSelectedCells,
+    intl,
   } = props;
 
+  const flowsheetColumns = useMemo(() => columns.map(column => ({ ...column, isResizable: false })), [columns]);
+  const pinnedColumns = flowsheetColumns.length ? [flowsheetColumns[0]] : [];
+  const overflowColumns = flowsheetColumns.length > 1 ? flowsheetColumns.slice(1) : [];
+
+  const flowsheetRows = useMemo(() => {
+    const noResultCellContent = (
+      <>
+        <span aria-hidden>{intl.formatMessage({ id: 'Terra.flowsheetDataGrid.no-result-display' })}</span>
+        <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.flowsheetDataGrid.no-result' })} />
+      </>
+    );
+
+    const newRows = [...rows];
+    newRows.forEach((row, rowIndex) => {
+      const newCells = [...row.cells];
+      newCells.forEach((cell, cellIndex) => {
+        // Cell has no content and is not a row header (first column), set content to "No result".
+        if (!cell.content && cellIndex !== 0) {
```

Comment:
Wouldn't this also display cells with a 0 value as No Result? 

That, along with things like `NaN` and `undefined` seem like this could be dangerous. A better approach might be to check for explicitly defined values like `null`, `'--'`, and `''`

Reply:
Good catch! A numeric 0 value does make this No Result. I can add explicit cases rather than a falsy check, assuming I know what exactly we want to check for. Are we not including `undefined` as a case to show No Result? Also, if the consumer explicitly passes a `'--'` do we want to treat it as No Result?

Reply:
I added a check for specifically `null`, `''`, and `'--'`.